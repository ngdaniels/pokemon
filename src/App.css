@font-face {
  font-family: 'Montserrat';
  src: local('Montserrat'), url(./Fonts/Montserrat-Light.ttf) format('truetype');
  font-display: swap;
}

@font-face {
  font-family: 'Montserrat';
  font-weight: bold;
  src: local('Montserrat'), url(./Fonts/Montserrat-SemiBold.ttf) format('truetype');
  font-display: swap;
}

body {
  font-family: 'Montserrat';
}

.title-bar {
  text-align: center;
  padding-top: 0.5rem;
  background-color: #222831;
  top: 0;
  position: fixed;
  width: 100%;
  z-index: 2;
}

.logo {
  height: 5rem;
  transition: 400ms;
}

.menu-bar {
  margin: 0;
  padding: 0;
  height: 5rem;
}

ul.menu-bar {
  background-color: #374045;
  padding: 0;
}

ul.menu-bar li {
  display: inline;
  list-style-type: none;
  margin: 0;
  text-align: center;
}

ul.menu-bar li a .menu-icon {
  margin-top: 1rem;
  height: 2rem;
}

ul.menu-bar li a {
  color: #FFF;
  text-decoration: none;
  padding: 0;
  display: inline-block;
  height: 100%;
  width: 50%;
}

.content {
  margin-top: 10.5rem;
  transition: 400ms;
}

.active {
  background-color: #30475e;
  font-weight: bold;
}

.text-center {
  text-align: center;
}

.bold {
  font-weight: bold;
}

.padding-top {
  padding-top: 0.5rem;
}

a {
  text-decoration: none;
  color: black;
}

input {
  border-radius: 1rem;
  font-family: 'Montserrat';
  font-weight: bold;
}

button {
  outline:none;
}

table {
  table-layout: fixed;
}

td.elipsis {
  white-space: normal;
  overflow: hidden;
  text-overflow: ellipsis;
}