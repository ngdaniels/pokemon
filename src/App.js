import React, { Component } from "react";
import $ from 'jquery';
import './App.css';
import Pokémon from './Assets/Pokémon.png';
import PokéDex from './Assets/PokéDex.png';
import Pokéball from './Assets/Pokéball.png';
import {
	Route,
	NavLink,
	Switch,
	HashRouter
} from "react-router-dom";
import List from './List.js';
import Detail from './Detail.js';
import Collection from './Collection.js';
import { gql } from '@apollo/client';
import client from './ApolloClient.js'

class App extends Component {
	constructor(props) {
		super(props);

		//Set initial storage
		if (!localStorage.getItem('owned')) {
			localStorage.setItem('owned', JSON.stringify([]));
		}

		this.state = {
			pokemons: [],
			isLoaded: false,
			fetching: false,
			offset: 0,
			error: null,
		};
	}

	fetchNext = () => {
		if (this.state.fetching) {
			return;
		}

		this.setState({
			fetching: true,
			isLoaded: false
		});

		//Fetch data
		client.query({
			query: gql`
				query pokemons($limit: Int, $offset: Int) {
					pokemons(limit: $limit, offset: $offset) {
						count
						next
						status
						results {
							id
							name
							image
						}
					}
				}
      		`,
			variables: {
				limit: 10,
				offset: this.state.offset,
			}
		})
			.then((res) => {
				this.setState({
					offset: this.state.offset + 10,
					pokemons: [...this.state.pokemons, ...res.data.pokemons.results],
					fetching: false,
					isLoaded: true
				});
			})
			.catch((error) => {
				this.setState({
					error
				});
			});
	}

	componentDidMount() {
		//Set event listener
		document.addEventListener('scroll', this.trackScrolling);

		//Fetch data
		this.fetchNext();
	};

	trackScrolling = () => {
		if ($(window).scrollTop() > 1) {
			$(".logo").css("height", "3rem");
		}
		else {
			$(".logo").css("height", "5rem");
		}
	};

	render() {
		return (
			<HashRouter>
				<div className="title-bar">
					<img src={Pokémon} className="logo" alt="logo" />
					<div>
						<ul className="menu-bar">
							<li><NavLink exact to="/"><img src={PokéDex} className="menu-icon" alt="pokedex-icon" /> <br></br> Pokémon List</NavLink></li>
							<li><NavLink to="/my_collection"><img src={Pokéball} className="menu-icon" alt="pokeball-icon" /> <br></br> My Pokémon</NavLink></li>
						</ul>
					</div>
				</div>

				<div className="content">
					<Switch>
						<Route exact path="/">
							<List pokemons={this.state.pokemons} error={this.state.error} isLoaded={this.state.isLoaded} isLoadedNext={this.state.isLoadedNext} fetchNext={this.fetchNext} />
						</Route>
						<Route path="/detail/:pokemonName" component={Detail} />
						<Route path="/my_collection/">
							<Collection />
						</Route>
					</Switch>
				</div>
			</HashRouter>
		);
	}
}

export default App;
