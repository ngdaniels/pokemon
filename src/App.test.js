import { render, screen } from '@testing-library/react';
import App from './App';

describe('App', () => {
	test('PokÃ©mon List', () => {
		var { asFragment } = render(<App/>);
        expect(asFragment()).toMatchSnapshot();
	});
});