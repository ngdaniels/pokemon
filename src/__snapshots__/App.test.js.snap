// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Pok√©mon List 1`] = `
<DocumentFragment>
  <div
    class="title-bar"
  >
    <img
      alt="logo"
      class="logo"
      src="Pok√©mon.png"
    />
    <div>
      <ul
        class="menu-bar"
      >
        <li>
          <a
            aria-current="page"
            class="active"
            href="#/"
          >
            <img
              alt="pokedex-icon"
              class="menu-icon"
              src="Pok√©Dex.png"
            />
             
            <br />
             Pok√©mon List
          </a>
        </li>
        <li>
          <a
            href="#/my_collection"
          >
            <img
              alt="pokeball-icon"
              class="menu-icon"
              src="Pok√©ball.png"
            />
             
            <br />
             My Pok√©mon
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="content"
  >
    <div>
      <div
        class="text-center"
        id="owned-search"
      >
        <p>
          Pok√©mons owned: 0
        </p>
        <form>
          <label>
            Search ¬†
            <input
              placeholder="Pok√©mon Number / Name"
              style="width: 65%; padding: 3px;"
              type="text"
              value=""
            />
            <input
              style="margin-left: -2rem; width: 2rem; padding: 0px;"
              type="submit"
              value="üîç"
            />
          </label>
        </form>
      </div>
      <div
        style="text-align: center; padding: 10px;"
      >
        <img
          alt="loading"
          class="css-3kjrfq"
          src="loading-pokeball.png"
          style="width: 20%; display: inline;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
